lib/pacemaker/pcmk_sched_resource.c

```diff
@@ -488,14 +484,19 @@ pcmk__assign_resource(pe_resource_t *rsc, pe_node_t *node, bool force)
     bool changed = false;
 
     if (rsc->children == NULL) {
+        pe_node_t *old = NULL;
+
         if (rsc->allocated_to != NULL) {
-            changed = true;
+            old = pe__copy_node(rsc->allocated_to);
```

Comment:
We could just save the old node details and compare to the new one. I'd prefer to use `pe__copy_node()` so that we can use `pe__same_node()` for clarity.